docs/lib/analytics/fragments/ios/identifyuser.md

```diff
@@ -2,33 +2,38 @@ This call sends information that you have specified about the user to Amazon Pin
 
 ```swift
 func identifyUser() {
-    let userState = AWSMobileClient.default().currentUserState
-    let getIdentityIdTask = AWSMobileClient.default().getIdentityId()
-    getIdentityIdTask.continueWith { (task) -> Any? in
-        if let error = task.error {
-            print("Failed to get identityId: \(error)")
-        }
 
-        guard let identityId = task.result else {
-            print("Missing identityId")
-            return nil
+    var identityId: String?
+    
+    // Get identityId
+    _ = Amplify.Auth.fetchAuthSession(listener: { event in
+        switch event {
+        case .success(let authSession):
+            let cognitoAuthSession = (authSession as? AuthCognitoIdentityProvider)?.getIdentityId()
+            switch cognitoAuthSession {
+            case .success(let identityId):
+                identityId = identityId
+            case .failure(let error):
+                print("Failed to get identity id: \(error)")
+            }
+        case .failure(let error):
+            print("Failed to fetch auth session: \(error)")
         }
+    })
```

Comment:
use `Amplify.Auth.getCurrentUser()` and `userId` instead

Reply:
The issue here is getCurrentUser is async call that when calling Amplify.Analytics.identifyUser(), AuthUser? which should be result haven't been returned yet. So the run will shows error "Fatal error: Unexpectedly found nil while unwrapping an Optional value". That's why I still put getIdentityId() here

Reply:
Oh, wait. Let me verify something

Reply:
The getCurrentUser() works, but only when user is signed in. So this Analytics docs should have a precondition something like "You should have a user signed in that can call this identifyUser()"

Reply:
cc: @TrekSoft @royjit

Reply:
Identityid - Amplify.fetchAuthSession, cast to CognitoIdentityProvider, get IdentityId for guest users
UserId - Cognito user pool user, must be authenticated to `getCurrentUser`

Reply:
i think we should just show how to identify a Cognito User Pool user for now, it's simple, and not async

Reply:
If the user is signed in through [Amplify.Auth.signIn](link to auth docs), then you can retrieve the current user and use their userId to identify the user with Analytics
```
guard let user = Amplify.Auth.getCurrentUser() else {
   print("User is not signed in")
   return
}

Amplify.Analytics.identifyUser(user.userId, withProfile: profile)
```

Reply:
Changed. I also tested it and it works. Please take a look